data-prepper-api/src/main/java/com/amazon/dataprepper/model/buffer/PayloadSizeOverflowException.java

```diff
@@ -0,0 +1,18 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.model.buffer;
+
+public class PayloadSizeOverflowException extends Exception {
```

Comment:
Perhaps we should use a different term rather than "payload." We don't use "payload" to refer to incoming messages into the buffer anywhere else in Data Prepper. Perhaps `ExceedsBufferCapacityException`, `MessageSizeOverflowException`, or `SizeOverflowException`.

Reply:
Sure. I will use `SizeOverflowException`.